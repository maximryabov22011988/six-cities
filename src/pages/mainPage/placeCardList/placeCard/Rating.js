import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  rating: PropTypes.number.isRequired
};

const MAX_RATING = 5;

const getRatingStyles = (rating) => {
  const ratingInPercent = (rating / MAX_RATING) * 100;
  return {
    width: `${ratingInPercent.toFixed(2)}%`
  };
};

function Rating(props) {
  const { rating } = props;

  return (
    <div className="place-card__rating rating">
      <div className="place-card__stars rating__stars">
        <span style={getRatingStyles(rating)} />
        <span className="visually-hidden">Rating</span>
      </div>
    </div>
  );
}

Rating.propTypes = propTypes;

export default Rating;
