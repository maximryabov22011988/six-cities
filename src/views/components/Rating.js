import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  parentClassName: PropTypes.string,
  rating: PropTypes.number.isRequired,
  showValue: PropTypes.bool
};

const MAX_RATING = 5;

const getRatingStyles = rating => {
  const ratingInPercent = (rating / MAX_RATING) * 100;
  return {
    width: `${ratingInPercent.toFixed(2)}%`
  };
};

function Rating({ parentClassName, rating, showValue }) {
  return (
    <div className={`${parentClassName}__rating rating`}>
      <div className={`${parentClassName}__stars rating__stars`}>
        <span style={getRatingStyles(rating)} />
        <span className="visually-hidden">Rating</span>
      </div>
      {showValue && (
        <span className={`${parentClassName}__rating-value rating__value`}>
          {rating}
        </span>
      )}
    </div>
  );
}

Rating.propTypes = propTypes;

export default Rating;
