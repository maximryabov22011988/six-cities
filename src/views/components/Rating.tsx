import * as React from 'react';

interface Props {
  isShowValue?: boolean;
  parentClassName: string;
  rating: number;
}

const MAX_RATING = 5;

const getRatingStyles = (rating) => {
  const ratingInPercent = (rating / MAX_RATING) * 100;
  return {
    width: `${ratingInPercent.toFixed(2)}%`,
  };
};

function Rating({ parentClassName, rating, isShowValue }: Props) {
  return (
    <div className={`${parentClassName}__rating rating`}>
      <div className={`${parentClassName}__stars rating__stars`}>
        <span style={getRatingStyles(rating)} />
        <span className="visually-hidden">Rating</span>
      </div>
      {isShowValue && <span className={`${parentClassName}__rating-value rating__value`}>{rating}</span>}
    </div>
  );
}

export default Rating;
