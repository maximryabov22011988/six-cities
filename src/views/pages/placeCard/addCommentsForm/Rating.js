import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

const RATING_TITLES = ['terribly', 'badly', 'not bad', 'good', 'perfect'];

function Rating() {
  return (
    <React.Fragment>
      {RATING_TITLES.reverse().map((title, index) => (
        <React.Fragment key={title}>
          <input
            className="form__rating-input visually-hidden"
            id={`${index + 1}-stars`}
            name="rating"
            type="radio"
            value={index + 1}
          />

          <label className="reviews__rating-label form__rating-label" htmlFor={`${index + 1}-stars`} title={title}>
            <svg className="form__star-image" height="33" width="37">
              <use xlinkHref="#icon-star" />
            </svg>
          </label>
        </React.Fragment>
      ))}
    </React.Fragment>
  );
}

Rating.propTypes = propTypes;
Rating.defaultProps = defaultProps;

export default Rating;
