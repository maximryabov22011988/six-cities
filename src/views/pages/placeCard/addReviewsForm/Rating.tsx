import * as React from 'react';


const propTypes = {
  rating: {
    currentRating: PropTypes.string,
    onChange: PropTypes.func.isRequired,
  },
  label: {
    ratingValue: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
  },
  radio: {
    checked: PropTypes.bool.isRequired,
    id: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
  },
};

const TERRIBLY = 'terribly';
const BADLY = 'badly';
const NOT_BAD = 'not bad';
const GOOD = 'good';
const PERFECT = 'perfect';

const RATING_MAP = {
  [TERRIBLY]: '1',
  [BADLY]: '2',
  [NOT_BAD]: '3',
  [GOOD]: '4',
  [PERFECT]: '5',
};

function Rating({ currentRating, onChange }) {
  return (
    <div className="reviews__rating-form form__rating">
      <Radio
        checked={currentRating === RATING_MAP[PERFECT]}
        id={`${RATING_MAP[PERFECT]}-stars`}
        value={RATING_MAP[PERFECT]}
        onChange={onChange}
      />
      <Label ratingValue={RATING_MAP[PERFECT]} title={PERFECT} />

      <Radio
        checked={currentRating === RATING_MAP[GOOD]}
        id={`${RATING_MAP[GOOD]}-stars`}
        value={RATING_MAP[GOOD]}
        onChange={onChange}
      />
      <Label ratingValue={RATING_MAP[GOOD]} title={GOOD} />

      <Radio
        checked={currentRating === RATING_MAP[NOT_BAD]}
        id={`${RATING_MAP[NOT_BAD]}-stars`}
        value={RATING_MAP[NOT_BAD]}
        onChange={onChange}
      />
      <Label ratingValue={RATING_MAP[NOT_BAD]} title={NOT_BAD} />

      <Radio
        checked={currentRating === RATING_MAP[BADLY]}
        id={`${RATING_MAP[BADLY]}-stars`}
        value={RATING_MAP[BADLY]}
        onChange={onChange}
      />
      <Label ratingValue={RATING_MAP[BADLY]} title={BADLY} />

      <Radio
        checked={currentRating === RATING_MAP[TERRIBLY]}
        id={`${RATING_MAP[TERRIBLY]}-stars`}
        value={RATING_MAP[TERRIBLY]}
        onChange={onChange}
      />
      <Label ratingValue={RATING_MAP[TERRIBLY]} title={TERRIBLY} />
    </div>
  );
}

Rating.propTypes = propTypes.rating;

function Label({ ratingValue, title }) {
  return (
    <label className="reviews__rating-label form__rating-label" htmlFor={`${ratingValue}-stars`} title={title}>
      <svg className="form__star-image" height="33" width="37">
        <use xlinkHref="#icon-star" />
      </svg>
    </label>
  );
}

Label.propTypes = propTypes.label;

function Radio({ checked, id, value, onChange }) {
  return (
    <input
      checked={checked}
      className="form__rating-input visually-hidden"
      id={id}
      name="rating"
      type="radio"
      value={value}
      onChange={onChange}
    />
  );
}

Radio.propTypes = propTypes.radio;

export default Rating;
