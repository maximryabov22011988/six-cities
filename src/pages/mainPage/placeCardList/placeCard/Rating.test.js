import React from 'react';
import renderer from 'react-test-renderer';

import Rating from './Rating';

it('Rating correctly renders', () => {
  const rating = renderer
    .create(
      <Rating
        rating={1}
      />
    )
    .toJSON();

  expect(rating).toMatchSnapshot();
});
