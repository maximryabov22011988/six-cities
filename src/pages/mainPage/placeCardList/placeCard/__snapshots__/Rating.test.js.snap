// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating correctly renders 1`] = `
<div
  className="place-card__rating rating"
>
  <div
    className="place-card__stars rating__stars"
  >
    <span
      style={
        Object {
          "width": "20.00%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;
